### PR TITLE
CORE-15813: Add REST endpoint for vnode connection string update

### PR DESCRIPTION
**Pull request by josephzunigadaly**
API PR: https://github.com/corda/corda-api/pull/1345

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -46,7 +46,7 @@ commonsTextVersion = 1.10.0
 bouncycastleVersion=1.76
 # Corda API libs revision (change in 4th digit indicates a breaking change)
 # Change to 5.2.0.xx-SNAPSHOT to pick up maven local published copy
-cordaApiVersion=5.2.0.10-beta+
+cordaApiVersion=5.2.0.10-alpha-1701255104685
```
Comment (josephzunigadaly) [inline gradle.properties]: Need to revert

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRestResource.kt [hunk 2]
```diff
@@ -41,6 +42,26 @@ interface VirtualNodeRestResource : RestResource {
         request: CreateVirtualNodeRequest
     ): ResponseEntity<AsyncResponse>
 
+    /**
+     * Requests an update to an existing virtual node.
+     *
+     * @throws `HttpApiException` If the request returns an exceptional response.
+     */
+    @HttpPUT(
+        path= "{virtualNodeShortId}",
```
Comment (vkolomeyko) [inline libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRestResource.kt]:
```suggestion
        path= "{virtualNodeShortId}/db",
```

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRestResource.kt [hunk 3]
```diff
@@ -41,6 +42,26 @@ interface VirtualNodeRestResource : RestResource {
         request: CreateVirtualNodeRequest
     ): ResponseEntity<AsyncResponse>
 
+    /**
+     * Requests an update to an existing virtual node.
```
Comment (vkolomeyko) [inline libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRestResource.kt]:
```suggestion
     * Requests an update to an existing virtual node database.
```

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/UpdateVirtualNodeRequest.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package net.corda.libs.virtualnode.endpoints.v1.types
+
+/**
+ * The data object sent via HTTP to request the update of a virtual node.
+ *
+ * @param shortHash The short hash of the virtual node.
+ */
+data class UpdateVirtualNodeRequest(
```
Comment (vkolomeyko) [inline libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/UpdateVirtualNodeRequest.kt]:
```suggestion
data class UpdateVirtualNodeDbRequest(
```

**Review by vkolomeyko — DISMISSED** [review]
LGTM

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**